### PR TITLE
Fix bugs and introduce user_tournament table

### PR DESCRIPTION
**Pull request by pmh192**

### CODE REVIEW

File: backend/src/main/java/com/tourneynizer/tourneynizer/dao/RosterDao.java [hunk 1]
```diff
@@ -18,13 +18,13 @@ public RosterDao(DataSource dataSource) {
         this.jdbcTemplate = new JdbcTemplate(dataSource);
     }
 
-    public void registerUser(User user, Team team, boolean isLeader) {
+    static void registerUser(User user, Team team, boolean isLeader, JdbcTemplate jdbcTemplate) {
```
Comment (lar-ryan-nick) [inline backend/src/main/java/com/tourneynizer/tourneynizer/dao/RosterDao.java]: Why is this package-private?

File: backend/src/main/java/com/tourneynizer/tourneynizer/dao/RosterDao.java [hunk 2]
```diff
@@ -41,8 +41,8 @@ public void registerUser(User user, Team team, boolean isLeader) {
         }
     }
 
-    public void registerUser(User user, Team team) {
-        registerUser(user, team, false);
+    static void registerUser(User user, Team team, JdbcTemplate jdbcTemplate) {
```
Comment (lar-ryan-nick) [inline backend/src/main/java/com/tourneynizer/tourneynizer/dao/RosterDao.java]: Again, why package-private?

Comment (pmh192) [reply backend/src/main/java/com/tourneynizer/tourneynizer/dao/RosterDao.java]: So only other Dao's have access to it.

File: backend/src/main/java/com/tourneynizer/tourneynizer/dao/RosterDao.java (hunk 1 above)
Comment (pmh192) [reply backend/src/main/java/com/tourneynizer/tourneynizer/dao/RosterDao.java]: 27999

File: backend/src/main/java/com/tourneynizer/tourneynizer/model/UserInfo.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+    public int wins, losses, tournaments, matches;
+
+    public UserInfo(int wins, int losses, int tournaments, int matches) {
+        this.wins = wins;
+        this.losses = losses;
+        this.tournaments = tournaments;
+        this.matches = matches;
+    }
+}
```
Comment (lar-ryan-nick) [inline backend/src/main/java/com/tourneynizer/tourneynizer/model/UserInfo.java]: Don't you also want to have a connecting user object or id value?

Comment (lar-ryan-nick) [inline backend/src/main/java/com/tourneynizer/tourneynizer/model/UserInfo.java]: Also how are you going to get all of this data at the same time?

Comment (pmh192) [reply backend/src/main/java/com/tourneynizer/tourneynizer/model/UserInfo.java]: User hasa UserInfo

Comment (pmh192) [reply backend/src/main/java/com/tourneynizer/tourneynizer/model/UserInfo.java]: Eventually they will be columns in user table, getting them on the fly would be too expensive.

File: backend/src/main/java/com/tourneynizer/tourneynizer/dao/RosterDao.java (hunk 1 above)
Comment (lar-ryan-nick) [reply backend/src/main/java/com/tourneynizer/tourneynizer/dao/RosterDao.java]: Why isn't a member function?

File: backend/src/main/java/com/tourneynizer/tourneynizer/model/UserInfo.java (hunk 3 above)
Comment (lar-ryan-nick) [reply backend/src/main/java/com/tourneynizer/tourneynizer/model/UserInfo.java]: Did you add it to the User class yet?

File: backend/src/main/java/com/tourneynizer/tourneynizer/dao/RosterDao.java (hunk 1 above)
Comment (pmh192) [reply backend/src/main/java/com/tourneynizer/tourneynizer/dao/RosterDao.java]: It'd be tricky for other DAOs to have other DAOs due to the nature of Spring

File: backend/src/main/java/com/tourneynizer/tourneynizer/model/UserInfo.java (hunk 3 above)
Comment (lar-ryan-nick) [reply backend/src/main/java/com/tourneynizer/tourneynizer/model/UserInfo.java]: But would sql query would give you all this info? Or are you gonna have to make multiple ones to make one single user info object?

Comment (lar-ryan-nick) [reply backend/src/main/java/com/tourneynizer/tourneynizer/model/UserInfo.java]: What sql query will give you all this info or will you have to make multiple queries to make one user info object?

File: backend/src/main/java/com/tourneynizer/tourneynizer/dao/RosterDao.java (hunk 1 above)
Comment (lar-ryan-nick) [reply backend/src/main/java/com/tourneynizer/tourneynizer/dao/RosterDao.java]: Ok then that makes sense

**Comment by lar-ryan-nick** [comment]
Looks great! Count this as an approval

**Pull request merged**